File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/java/Fibonacci.java
import java.io.*;
import java.lang.*;
import java.util.*;
import java.math.*;


class Fibonacci {
    
    /*@ public normal_behavior
      @ ensures \result == ((x % 4294967296L + 6442450944L) % 4294967296L) - 2147483648L;
      @ model public static pure \bigint fb_wrap(\bigint x);
      @*/
    /*@ public normal_behavior
      @ requires n >= 0;
      @ ensures \result == (n <= 1 ? n : fb_value(n - 1) + fb_value(n - 2));
      @ measured_by n;
      @ model public static pure \bigint fb_value(\bigint n);
      @*/
    /*@ public normal_behavior
      @ requires n >= 0;
      @ assignable \nothing;
      @ ensures \result == fb_wrap(fb_value(n));
      @ measured_by n;
      @*/
    //@ code_java_math
    public static int fibonacci(int n) {
        if (n == 0) {
            return 0;
        } else if (n == 1) {
            return 1;
        } else {
            return fibonacci(n - 1) + fibonacci(n - 2);
        }
    }
}

File: FormalBench-data/FilteredData/specifications/seed_726_per_category_10_653ade686f/c/Fibonacci.c
#include "java_arrays.h"
#include <stdint.h>

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left - right);
*/
static int32_t java_sub(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left - (uint32_t)right);
}

/*@
  assigns \nothing;
  ensures \result == (int32_t)((integer)left + right);
*/
static int32_t java_add(int32_t left, int32_t right) {
    return (int32_t)((uint32_t)left + (uint32_t)right);
}

/*@
  logic integer fb_value(integer n) = n <= 1 ? n : fb_value(n - 1) + fb_value(n - 2);
*/
/*@
  requires n >= 0;
  decreases n;
  assigns \nothing;
  ensures \result == (int32_t)fb_value(n);
*/
int32_t fibonacci(int32_t n) {
    if (n == 0) {
        return 0;
    } else if (n == 1) {
        return 1;
    }
    int32_t left = fibonacci(java_sub(n, 1));
    int32_t right = fibonacci(java_sub(n, 2));
    return java_add(left, right);
}
